packages/explorable-explanations/src/modules/auctions.js

```diff
@@ -320,37 +446,43 @@ auction.draw = async (index) => {
   }
 
   for (const step of steps) {
-    if (window.cancelPromise || window.cancelPromiseForPreviousAndNext) {
-      return;
-    }
+    PromiseQueue.nextStepSkipIndex.push(PromiseQueue.queue.length - 1);
+    PromiseQueue.add(async () => {
+      const { component, props, callBack } = step;
 
-    const { component, props, callBack } = step;
-    const returnValue = await component(props); // eslint-disable-line no-await-in-loop
-    const delay = component === Box ? 1000 : 0;
+      const returnValue = await component(props); // eslint-disable-line no-await-in-loop
 
-    if (props?.title === 'Load Interest Group') {
-      await bubbles.barrageAnimation(index); // eslint-disable-line no-await-in-loop
-    }
+      const delay = component === Box ? 1000 : 0;
 
-    if (props?.title === 'generateBid()') {
-      const x = props.x();
-      const y = props.y();
+      if (props?.title === 'Load Interest Group') {
+        await bubbles.barrageAnimation(index); // eslint-disable-line no-await-in-loop
+        await utils.delay(500); // eslint-disable-line no-await-in-loop
 
-      // eslint-disable-next-line no-await-in-loop
-      await RippleEffect({
-        x: x + config.flow.box.width + 2,
-        y: y + config.flow.box.height / 2,
-      });
-    }
+        utils.wipeAndRecreateInterestCanvas(); // eslint-disable-line no-await-in-loop
+      }
 
-    if (callBack) {
-      callBack(returnValue);
-    }
-    await utils.delay(delay); // eslint-disable-line no-await-in-loop
+      if (props?.title === 'generateBid()') {
```

Comment:
As we have to check conditions and text can change in future, let's add keys to the object for comparison.

Reply:
Resolved now using `extraProps` object instead of title for comparison.